src/main/java/com/emt/service/MatchService.java

```diff
@@ -19,11 +23,13 @@
 
 @Service
 @RequiredArgsConstructor
+@Transactional
 public class MatchService {
 
   private static final BigDecimal CONSTANT_K = new BigDecimal("30");
   private final MatchRepository matchRepository;
   private final MatchMapper matchMapper;
+  private final PlayerRepository playerRepository;
```

Comment:
It's bad practice to inject a repository into unrelated services, call it through a service instead